azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/DeployMojo.java

```diff
@@ -45,40 +52,57 @@
 
     @Override
     protected void doExecute() throws Exception {
-        createOrUpdateWebApp();
+        final ConfigurationParser parser = getParserBySchemaVersion();
+        final WebAppConfiguration webAppConfig = parser.getWebAppConfiguration();
+        createOrUpdateWebApp(webAppConfig);
         deployArtifacts();
     }
 
-    protected void createOrUpdateWebApp() throws Exception {
+    protected ConfigurationParser getParserBySchemaVersion() throws MojoExecutionException {
+        final String schemaVersion = StringUtils.isEmpty(getSchemaVersion()) ? "v1" : getSchemaVersion();
+
+        switch (schemaVersion.toLowerCase(Locale.ENGLISH)) {
+            case "v1":
+                return new V1ConfigurationParser(this);
+            case "v2":
+                return new V2ConfigurationParser(this);
+            default:
+                throw new MojoExecutionException(SchemaVersion.UNKNOWN_SCHEMA_VERSION);
+        }
+    }
+
+    protected void createOrUpdateWebApp(final WebAppConfiguration config) throws Exception {
         final WebApp app = getWebApp();
         if (app == null && this.isDeployToDeploymentSlot()) {
             throw new MojoExecutionException(WEBAPP_NOT_EXIST_FOR_SLOT);
         }
+        // todo: use parser to getAzureClient from mojo
+        final RuntimeHandler runtimeHandler = getFactory().getRuntimeHandler(config, getAzureClient(), getLog());
         if (app == null) {
-            createWebApp();
+            createWebApp(runtimeHandler);

```

Comment:
Can we put `create` and `update` logic into runtimHandler?

If it is possible, you can do this change in next PR

Reply:
Yes, I was considering to refactor the create and update logic, if we are handling it in the runtime, it needs to know the web app, or at least if create is necessary.